lib/paper_trail/reifier.rb

```diff
@@ -0,0 +1,256 @@
+module PaperTrail
+
+  # Given a version record and some options, builds a new model object.
+  # @api private
+  class Reifier
+
+    # @api private
+    def initialize(version, options)
+      @options = options.dup
+      @version = version
+    end
+
+    # See `VersionConcern#reify` for documentation.
+    # @api private
```

Comment:
I see, thanks for the explanation. Makes sense to me.  Sorry I've been slow on reviewing this, just started a new job, I'll try to go through it more thoroughly in the next few days, looks good on the surface.